security/fma-supervisor.md

```diff
@@ -0,0 +1,269 @@
+- Risk Assessment
+    - Low Impact, High likelihood.
+    - Software fails; ports become blocked. At some point an interruption will take down a Supervisor.
+    - We have redundancy solutions being designed [here](https://github.com/ethereum-optimism/design-docs/pull/218/files?short_path=88594e4#diff-88594e47f0a70261441a7452448ef1f240c7c0f15b9132c7789b2ee2d0e07bd2), which will make Supervisor failures less impactful for Chain Operators.
+    - Other Node Operators can adopt similar redundancy measures, but may follow their own infrastructure designs . Operators should treat the Supervisor as consensus critical, and manage it similar to how they would manage their L1 source.
+
+## FM1b: Supervisor Is Corrupted
+
+- Description
+    - Not only is the supervisor unavailable, its database has been corrupted or destroyed.
+    - In this scenario, all of the above applies, and the recovery is delayed by our ability to restore a working Supervisor.
+- Risk Assessment
+    - Low Impact, Low likelihood.
+    - The Supervisor has robust database trimming to clear invalid states and partial writes.
+    - This code, like all Supervisor code, could be tested more.
+- Mitigations
+    - There is a feature of the Supervisor that databases can be replicated over HTTP. If a Supervisor needs to be quickly synced it can bootstrap its databases from an existing Supervisor.
+    - Supervisors themselves should be made redundant, so if one goes down, another may serve in its place. Supervisor *replacement* is not something that is well tested, and more likely you’d switch to use that backup Supervisor *and* its Managed Nodes.
+
+## FM2a: Supervisor Incorrectly Determines Message Validity in Response to Sequencer - Unsafe Block Only
+
+- Description
+    - If the Supervisor were to claim an Executing Message is valid when it actually isn’t, Sequencers may use this determination to include an invalid message in their local-unsafe chain.
+    - Assuming the Supervisor correctly evaluates the cross-promotion (a later step), it will experience a cross-unsafe head stall at this error point, because of the cross-invalid message.
+    - The Sequencer does not care about cross-unsafe chain stalling, because unsafe data is subject to change, and as far as it knows, all the messages it has included are valid, so the *expectation* is that other dependent data will unstick it.
+    - Eventually, the block is published to the L1 and becomes local-safe data.
+    - When the Supervisor attempts to promote the local-safe data to cross-safe, it discovers the invalid message and issues an invalidation and replacement.
+    - The Replacement block is applied to the block which contained the invalid message, and the chain has now reorg’d out all blocks from the invalid message to the safe head (effectively resetting the chain back to the stalled cross-unsafe head).
+- Risk Assessment
+    - Medium Impact, Medium Likelihood.
```

Comment:
If a third party bridge or centralized exchange trusts unsafe interop messages, this is a risk. Dangerous scenario:
1. Chain A includes initiating message for "move 1000 ETH to Chain B"
2. Chain B sees the unsafe initiating message
3. Executing message to mint 1000 ETH on Chain B is executed
4. Attacker moves 1000 ETH out to L1 via a third party bridge or CEX which just trusts the unsafe head
5. An unsafe chain reorg happens on Chain A, and the initiating message is reorged out (this could happen maliciously, or even via just an L1 reorg)
6. Attacker has stolen 1000 ETH


Mitigation: Ensure that any applications which enabling moving substantial value *out* of the Superchain understand this risk, and ideally require making sure the chain is safe

Reply:
Minor clarification - an L1 reorg is actually not a vector for this, because the unsafe chain isn't based in any L1. And from the other perspective, any L1 reorg which changes initiating messages *must also* change the executing messages downstream from it. So L1 reorgs apply pretty transparently to the interop protocol.